src/miniscript/decode.rs

```diff
@@ -134,7 +134,9 @@ pub enum Terminal<Pk: MiniscriptKey, Ctx: ScriptContext> {
     /// `<key>`
     PkK(Pk),
     /// `DUP HASH160 <keyhash> EQUALVERIFY`
-    PkH(Pk::Hash),
+    PkH(Pk),
+    /// Only for Parsing PkH
+    RawPkH(Pk::Hash),
```

Comment:
In 36ff0d2bf8b9df789a5581f6a10f9f4457803331:

Nit: could say "only for parsing PkH from Script"